nexus/reconfigurator/planning/src/blueprint_editor/sled_editor.rs

```diff
@@ -135,6 +354,60 @@ impl SledEditor {
         }
     }
 
+    fn validate_decommisionable(&self) -> Result<(), SledEditError> {
+        // TODO-john The disks and datasets checks below don't pass what the
+        // planner does currently to decommission sleds: if a sled is expunged,
+        // we'll omit its disks and datasets from the outgoing blueprint
+        // entirely without setting them all to the `Expunged` disposition.
+        // Fixing this will conflict with ongoing disk work, so for now these
+        // checks are commented out.
+        /*
+        // Check that all disks are expunged...
+        if let Some(disk) =
+            self.disks(DiskFilter::All).find(|disk| match disk.disposition {
+                BlueprintPhysicalDiskDisposition::InService => true,
+                BlueprintPhysicalDiskDisposition::Expunged => false,
+            })
+        {
+            return Err(SledEditError::NonDecommissionableDiskInService {
+                disk_id: disk.id,
+                zpool_id: disk.pool_id,
+            });
+        }
+
+        // ... and all datasets are expunged ...
+        if let Some(dataset) =
+            self.datasets(BlueprintDatasetFilter::All).find(|dataset| {
+                match dataset.disposition {
+                    BlueprintDatasetDisposition::InService => true,
+                    BlueprintDatasetDisposition::Expunged => false,
+                }
+            })
+        {
+            return Err(SledEditError::NonDecommissionableDatasetInService {
+                dataset_id: dataset.id,
+                kind: dataset.kind.clone(),
+            });
+        }
+        */
```

Comment:
You knew I was going to suggest putting this into an issue instead. 😄 It could be in a comment for an existing issue that covers the work that would introduce these checks or a standalone issue we track under Reconfigurator.

Reply:
Removed in 79535ec18 and opened #7238